src/main/java/pages/FlightSearchPage.java

```diff
@@ -0,0 +1,228 @@
+    private WebElement nextButtonRight;
+
+    @FindBy(css = ".drp-calendar.left .loading")
+    private WebElement calendarLoading;
+
+    @FindBy(css = "div.passengers__text")
+    private WebElement passengersButton;
+
+    @FindBy(css = "div.passengers__popover.passengers__popover--active")
+    private WebElement passengerPopover;
+
+    @FindBy(css = "div.passengers__popover--active .choosePassengers")
+    private WebElement applyPassengersButton;
+
+    @FindBy(css = "span.currency-selected")
+    private WebElement currencyDropdown;
+
+    @FindBy(css = "div.dropdown-menu.show")
+    private WebElement dropdownMenu;
+
+    @FindBy(id = "inputadult")
+    private WebElement adultInput;
+
+    @FindBy(css = "#inputadult ~ button.quantity__plus")
+    private WebElement adultPlusButton;
+
+    @FindBy(id = "inputchild")
+    private WebElement childInput;
+
+    @FindBy(css = "#inputchild ~ button.quantity__plus")
+    private WebElement childPlusButton;
+
+    @FindBy(id = "inputbaby")
+    private WebElement babyInput;
+
+    @FindBy(css = "#inputbaby ~ button.quantity__plus")
+    private WebElement babyPlusButton;
+
+    @FindBy(css = "div.dropdown-menu.show input.form-control")
+    private WebElement arrivalSearchBox;
+
+    @FindBy(css = "ul.dropdown-menu.inner.show")
+    private WebElement cityDropdownList;
+
+    @FindBy(css = "input[data-note='Выберите дату обратно']")
+    private WebElement returnInput;
+
+    private static final String CITY_DROPDOWN_ITEM = "//ul[contains(@class,'dropdown-menu')]//span[contains(text(), '%s')]";
+    private static final String CALENDAR_DATE_CELL = ".//td[not(contains(@class, 'off')) and normalize-space(text())='%s']";
+    private static final String CURRENCY_OPTION = "//span[@class='only-code' and text()='%s']";
+
+    public FlightSearchPage(WebDriver driver) {
+        super(driver);
+    }
+
+    private By getCityDropdownItem(String city) {
+        return By.xpath(String.format(CITY_DROPDOWN_ITEM, city));
+    }
+
+    private By getCalendarDateCell(String day) {
+        return By.xpath(String.format(CALENDAR_DATE_CELL, day));
+    }
+
+    private By getCurrencyOption(String currency) {
+        return By.xpath(String.format(CURRENCY_OPTION, currency));
+    }
+
+    public void enableRoundTripIfDisabled() {
+        waitUntilClickable(toggleLabel);
+        if (oneWayToggleCheckbox.isSelected()) {
+            toggleLabel.click();
+            waitUntilReturnInputVisible();
+        }
+    }
+
+    public void selectDepartureCity(String cityName) {
+        fromButton.click();
+        waitUntilVisible(cityDropdownList);
+        WebElement city = waitUntilClickable(getCityDropdownItem(cityName));
+        scrollIntoView(city);
+        city.click();
+    }
+
+    public void selectArrivalCity(String cityName) {
+        waitUntilClickable(toButton).click();
+        try {
+            WebElement searchBox = wait.until(ExpectedConditions.presenceOfElementLocated(ARRIVAL_SEARCH_BOX));
```

Comment:
Please create a webElement for ARRIVAL_SEARCH_BOX using PageFactory. You can use WebElement object for waiters as well, you will just need to use a different expected condition